ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/plugin/ExecutionEnvironmentSection.java

```diff
@@ -246,8 +254,12 @@ public void run() {
 	private void updateButtons() {
 		Table table = fEETable.getTable();
 		TablePart tablePart = getTablePart();
-		tablePart.setButtonEnabled(0, isEditable());
-		tablePart.setButtonEnabled(1, isEditable() && table.getSelection().length > 0);
+		tablePart.setButtonEnabled(0,
+				isEditable() && table.getItemCount() == 1 && (table.getItem(0).getText().startsWith("JRE") //$NON-NLS-1$
+						|| table.getItem(0).getText().startsWith("JavaSE") //$NON-NLS-1$
+						|| table.getItem(0).getText().startsWith("J2SE"))); //$NON-NLS-1$
```

Comment:
I think Its better to have below condition for `Change` button

`tablePart.setButtonEnabled(0, isEditable() && table.getSelectionCount() == 1);`

Reply:
That would make the implementation more difficult. At the moment you can just replace the first (and only) EE `existingEEs.get(0)` in `handleChange()`. Furthermore then the question would arise when to update the Classpath? Only if the EE corresponding to the java version configured in the classpath/compiler settings?
Because the vast majority of Plugin/Bundle projects has only one EE I suggest to keep this button for that simple case and let those that have complex EE configs handle everything on their own.